labextension/src/widgets/LeftPanel.tsx

```diff
@@ -274,7 +289,23 @@ export class KubeflowKaleLeftPanel extends React.Component<IProps, IState> {
           }
           experiment_name = notebookMetadata['experiment_name'];
         } else {
-          this.resetState();
+          // If no experiment data exists, use the first experiment from the list if available
+          if (this.state.experiments.length > 0) {
+            experiment = this.state.experiments[0];
+            experiment_name = this.state.experiments[0].name;
+          } else {
+            // Keep existing experiment values if they exist, otherwise use defaults
+            if (
```

Comment:
I wonder if it wouldn't be easier to setup the experiment default value on line 260/257:

```
      // if the key exists in the notebook's metadata
      if (notebookMetadata) {
        let experiment: IExperiment = this.state.metadata.experiment;
        let experiment_name: string = this.state.metadata.experiment_name;
        if (notebookMetadata['experiment']) {
          experiment = {
            id: notebookMetadata['experiment']['id'] || this.state.metadata.experiment.id,
            name: notebookMetadata['experiment']['name'] || this.state.metadata.experiment.name
          };
          experiment_name = experiment.name;
```

With this I would say we should be able to avoid the code boilerplate.